src/ert/analysis/_es_update.py

```diff
@@ -238,168 +249,111 @@ def _create_temporary_parameter_storage(
 
 
 def _get_obs_and_measure_data(
-    obs: "EnkfObs",
+    obs: EnkfObs,
     source_fs: EnsembleReader,
     selected_observations: List[Tuple[str, List[int]]],
     ens_active_list: List[int],
-) -> Tuple[DataFrame, DataFrame]:
-    data_keys = defaultdict(set)
-    obs_data = []
-    t_summary = 0.0
-    t_gendata = 0.0
-    _logger.debug("_get_obs_and_measure_data() - start")
-    for obs_key, active_list in selected_observations:
-        obs_vector = obs[obs_key]
+) -> Tuple[
+    npt.NDArray[np.float_],
+    npt.NDArray[np.float_],
+    npt.NDArray[np.float_],
+    npt.NDArray[np.str_],
+]:
+    measured_data = []
+    observation_keys = []
+    observation_values = []
+    observation_errors = []
+    for obs_key, obs_active_list in selected_observations:
+        group, observation = obs.get_dataset(obs_key)
+        if obs_active_list:
+            index = observation.coords.to_index()[obs_active_list]
+            sub_selection: DefaultDict[str, List[str]] = defaultdict(list)
+            for row in index:
```

Comment:
Think we can get rid of this for loop by using a `set` to remove duplicates.
Something like this:

`sub_selection = {name: list(set(index.get_level_values(name))) for name in index.names}`

I can't seem to find a test that hits this code, i.e., a test where `obs_active_list` is `True`.

Reply:
Probably true, I will try to write one.